ROS/controls/src/moteus_wrapper.py

```diff
@@ -0,0 +1,212 @@
+        current_max_lane_id = -1
+
+        # We build all of data from the mapping at once
+        # instead of using list/dictionary comprehensions
+        # to reduce the number of iterations through the list
+        for motor_id, can_bus_lane_id in motor_list.items():
+            # Determines if we need to add more lanes
+            if can_bus_lane_id > current_max_lane_id:
+                # Add (can_bus_lane_id - current_max_lane_id) more lanes
+                moteus_ids += [[] for _ in range(current_max_lane_id, can_bus_lane_id)]
+
+            moteus_ids[can_bus_lane_id].append(motor_id)
+
+        # Note that the call to the initializer will block
+        self.moteus_instance = Moteus(moteus_ids, simulation=True)
+
+        self.desired_orientation_sub = rospy.Subscriber(
+            constants.CONTROLS_PKG_DESIRED_ORIENTATION_TOPIC,
+            MotorOrientation,
+            callback=self.__desired_orientation_callback,
+            queue_size=constants.CONTROLS_PKG_SUBSCRIBER_QUEUE_SIZE
+        )
+
+        self.current_orientation_publisher = rospy.Publisher(
+            constants.CONTROLS_PKG_CURRENT_ORIENTATION_TOPIC,
+            MotorOrientation,
+            queue_size=constants.CONTROLS_PKG_PUBLISHER_QUEUE_SIZE
+        )
+
+    # MARK: - Methods -
+
+    def motor_description_for_id(self, hwid: int):
+        """Fetches the CAN bus lane ID corresponding to the
+        motor with ID _id_
+
+        @param hwid: The hardware ID identifying the motor whose
+        descriptor should be fetched
+
+        @return: An identifier naming a distinct CAN bus lane
+        """
+        return self.id_map[hwid]
+
+    # MARK: - ROS Messages -
+
+    def run_motor_orientation_loop(self):
+        """Starts the infinite loop to continuously
+        publish ROS notifications about the current state
+        of the Moteus motors
+
+        This method puts the program in an infinite
+        loop, during each iteration of which the node publishes
+        the current orientation of the Moteus motors
+        """
+        rate = rospy.Rate(self.hw_refresh_rate)
+
+        # Wait until the motors are ready (if they aren't already)
+        self.moteus_instance.waitUntilReady()
+
+        while (not rospy.is_shutdown()):
+            latest_motor_state = self.moteus_instance.getParsedResults()
+            motor_states = []
+
+            # Each 'motor_desc' is a dictionary corresponding
+            # to the state of a motor
+            for motor_desc in latest_motor_state:
+                motor_state = MotorState()
+                motor_state.position = motor_desc['POSITION']
+                motor_state.velocity = motor_desc['VELOCITY']
+                motor_state.torque = motor_desc['TORQUE']
+                motor_states.append(motor_state)
+
+            msg = MotorOrientation(numMotors=self.num_motors, states=motor_states)
+            self.current_orientation_publisher.publish(msg)
+            rate.sleep()
```

Comment:
calling rate.sleep() here may impact performance. It may be nessicary to sync the publishing/receiving of messages to prevent message backup, just something to be aware of.